internal/user/user_profile_test.go

```diff
@@ -7,38 +7,82 @@ import (
 	"github.com/kopia/kopia/internal/user"
 )
 
+func TestLegacyUserProfile(t *testing.T) {
+	p := &user.Profile{
+		PasswordHashVersion: 1, // hashVersion1
+	}
+
+	if p.IsValidPassword("bar") {
+		t.Fatalf("password unexpectedly valid!")
+	}
+
+	p.SetPassword("foo")
```

Comment:
nit; IFF you go down the route of changing the user-profile algorithm settings after SetPssword() then lets also assert that `p.PasswordHashVersion` is cleared and `KeyDerivationAlgorithm` is set correctly.